embassy-stm32f4/src/exti.rs

```diff
@@ -109,18 +99,20 @@ impl<T: HalExtiPin + WithInterrupt + 'static> WaitForRisingEdge for ExtiPin<T> {
         s.pin.clear_interrupt_pending_bit();
         async move {
             let fut = InterruptFuture::new(&mut s.interrupt);
-            let mut exti: EXTI = unsafe { mem::transmute(()) };
+            cortex_m::interrupt::free(|_| {
+                let mut exti: EXTI = unsafe { mem::transmute(()) };
 
-            s.pin.trigger_on_edge(&mut exti, Edge::RISING);
-            s.pin.enable_interrupt(&mut exti);
+                s.pin.trigger_on_edge(&mut exti, Edge::RISING);
+                s.pin.enable_interrupt(&mut exti);
+            });
```

Comment:
error[E0500]: closure requires unique access to `s` but it is already borrowed
   --> embassy-stm32f4\src\exti.rs:102:39
    |
101 |             let fut = InterruptFuture::new(&mut s.interrupt);
    |                                            ---------------- borrow occurs here
102 |             cortex_m::interrupt::free(|_| {
    |                                       ^^^ closure construction occurs here
...
105 |                 s.pin.trigger_on_edge(&mut exti, Edge::RISING);
    |                 - second borrow occurs due to use of `s` in closure
...
108 |             fut.await;
    |             --- first borrow later used here